lib/sinon/proxy-invoke.js

```diff
@@ -2,18 +2,20 @@
 
 const arrayProto = require("@sinonjs/commons").prototypes.array;
 const proxyCallUtil = require("./proxy-call-util");
+const globalContext = require("./util/core/global-context");
 
 const push = arrayProto.push;
 const forEach = arrayProto.forEach;
 const concat = arrayProto.concat;
 const ErrorConstructor = Error.prototype.constructor;
 const bind = Function.prototype.bind;
 
-let callId = 0;
-
-module.exports = function invoke(func, thisValue, args) {
+module.exports = function invoke(func, thisValue, args, ctx = globalContext) {
     const matchings = this.matchingFakes(args);
-    const currentCallId = callId++;
+    if (ctx.callId == null) {
+        ctx.callId = 0;
+    }
```

Comment:
When does this ever happen? I cannot see any tests exercising this.

I see the fallback global module, where it is already initialised, and I see the sandbox, which is also initialised.